src/Npgsql/NpgsqlSchema.cs

```diff
@@ -563,6 +566,247 @@ static DataTable GetDataSourceInformation(NpgsqlConnection conn)
+            // We populate the DataType above from mapping.ClrTypes, which means we take the .NET type from
+            // which we *infer* the PostgreSQL type. Since only a single PostgreSQL type gets inferred from a given
+            // .NET type, we never have the same DataType in more than one row - so the mapping is always the
+            // best match. See the hardcoding override  below for some exceptions.
+            row["IsBestMatch"] = true;
+            row["IsCaseSensitive"] = true;
+            row["IsConcurrencyType"] = false;
+            row["IsFixedLength"] = false;
+            row["IsFixedPrecisionAndScale"] = false;
+            row["IsLiteralSupported"] = false;  // See hardcoding override below
+            row["IsLong"] = false;
+            row["IsNullable"] = true;
+            row["IsSearchable"] = true;
+            row["IsSearchableWithLike"] = false;
+            row["IsUnsigned"] = DBNull.Value; // See hardcoding override below
+            // LiteralPrefix: no literal for now except for strings, see hardcoding override below
+            // LiteralSuffix: no literal for now except for strings, see hardcoding override below
+            row["MaximumScale"] = DBNull.Value;
+            row["MinimumScale"] = DBNull.Value;
+            // NativeDataType is unset
+            row["OID"] = type.OID;
+        }
+
+        /// <summary>
+        /// Sets some custom, hardcoded info on a DataType row that cannot be loaded/inferred from PostgreSQL
+        /// </summary>
+        static void PopulateHardcodedDataTypeInfo(DataRow row, PostgresType type)
+        {
+            if (type.Name == "text" || type.Name == "varchar" || type.Name == "char")
+            {
+                row["CreateFormat"] += "({0})";
+                row["CreateParameters"] = "size";
+                row["IsSearchableWithLike"] = true;
+                row["IsLiteralSupported"] = true;
+                row["LiteralPrefix"] = "'";
+                row["LiteralSuffix"] = "'";
+
+                if (type.Name == "varchar" || type.Name == "char")
+                {
+                    row["DataType"] = "String";
+                    row["IsBestMatch"] = false;
+                }
+            }
+
+            if (type.Name == "numeric")
+            {
+                row["CreateFormat"] += "({0},{1})";
+                row["CreateParameters"] = "precision, scale";
+                row["MaximumScale"] = 16383;
+                row["MinimumScale"] = 16383;
+            }
+
+            if (type.Name == "bytea")
+                row["IsLong"] = true;
+
+            if (type.Name == "citext")
+                row["IsCaseSensitive"] = false;
+
+            if (type.Name == "integer" || type.Name == "smallint" || type.Name == "bigint" ||
+                type.Name == "double precision" || type.Name == "real" ||
+                type.Name == "numeric" || type.Name == "money")
+            {
+                row["IsUnsigned"] = false;
```

Comment:
Could we lift all of these keys to `const` fields or something similar? It would make jumping between different assignment locations easier in most IDEs, as well as guard against typos during future modifications.

Reply:
What do you have in mind? I'd rather keep the constants inline, rather than extract them and use them in a single place... IMHO it makes things more readable...

Reply:
Nothing more interesting than a bad habit of typing "unsinged" in place of "unsigned" :)

The inline literals do make it easier to skim, so disregard my earlier comment.

Reply:
OK :)

I'll just mention that you can just as well accidentally type `Unsinged` for a constant name or value so I don't think this would improve much... Thanks for the comments in any case.